include/networkit/numerics/LAMG/Lamg.hpp

```diff
@@ -243,6 +269,46 @@ void Lamg<Matrix>::parallelSolve(const std::vector<Vector>& rhs, std::vector<Vec
     }
 }
 
+template<class Matrix>
+void Lamg<Matrix>::parallelSolve(const RHSLoader& rhsLoader,
+                                 const ResultProcessor& resultProcessor,
+                                 std::pair<count, count> rhsSize,
+                                 count maxConvergenceTime,
+                                 count maxIterations) {
+    if (numComponents == 1) {
+        const index numThreads = omp_get_max_threads();
+        if (compSolvers.size() != numThreads) {
+            compSolvers.clear();
+
+            for (index i = 0; i < (index) numThreads; ++i) {
+                compSolvers.push_back(SolverLamg<Matrix>(compHierarchies[0], smoother));
+            }
+        }
+
+        count n = rhsSize.first;
+        count m = rhsSize.second;
+        std::vector<Vector> results(numThreads, Vector(m));
+        std::vector<Vector> RHSs(numThreads, Vector(m));
```

Comment:
solvers are responsible for resources. An alternative would be to give this control to user, but I don't see much benefit in this